pkg/daemon/update.go

```diff
@@ -426,15 +427,14 @@ func (dn *Daemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newConfig
 
 }
 
-func calculatePostConfigChangeActionFromFileDiffs(oldIgnConfig, newIgnConfig ign3types.Config) (actions []string) {
+func calculateConfigChangeActionFromFileDiffs(oldIgnConfig, newIgnConfig ign3types.Config) (drain bool, actions []string, err error) {
```

Comment:
Same as in other func, why add drain as a bool instead of to actions?

Reply:
I think keeping them separate is more in line with Sinny's comment. Drain is handled separately from actions (before writing files), while actions are handled after writing files by `performPostConfigChangeActions`. Mixing drain in with args to `performPostConfigChangeActions` seems to unnecessarily intertwine the logic, when drain is really a pre-config-change-action.

Reply:
yeah i think my point is that if it can't be neatly packaged into this function, then it probably doesn't belong there as having a bool return is unexpected and inconsistent.

Reply:
So would you prefer returning either `["drain"]` or `[]`? Why is consistency between pre and post actions important? Are we expecting to ever have more pre actions? Do you think the last solution I proposed would be okay?

Reply:
I'll defer to you and @sinnykumari on choosing which option is better between the 2, but I think they are both preferable to current implementation :+1: 

ill close this out and let you two resolve elsewhere